src/plugins/diagram-viewer/diagram-registration.ts

```diff
@@ -3,10 +3,16 @@ import { kDiagramDefaultHeight, kDiagramToolID } from "./diagram-types";
 import { defaultDiagramContent, DiagramContentModel } from "./diagram-content";
 import { DiagramToolComponent } from "./diagram-tool";
 import DiagramToolIcon from "./src/assets/program.svg";
+import { DiagramMigrator } from "./diagram-migrator";
 
 registerToolContentInfo({
   id: kDiagramToolID,
-  modelClass: DiagramContentModel,
+  // TODO: maybe there is there a better way to do this kind of casting?
+  //   The issue is that modelClass prop has a type of `typeof ToolContentModel`,
+  //   That type is pretty restrictive and doesn't accommodate the return of 
+  //   types.snapshotProcessor. There might be a better way to get a 
+  //   typescript type for a MST "Class" which is less restrictive
+  modelClass: DiagramMigrator as typeof DiagramContentModel,
```

Comment:
This kind of issue crops up every time I try to use `snapshotProcessor`. The semantics are just wrong. This is why I continue to use the deprecated `preProcessSnapshot`, which is simpler to code and has the proper semantics.